introspection-engine/connectors/sql-introspection-connector/src/re_introspection.rs

```diff
@@ -322,18 +322,25 @@ pub fn enrich(old_data_model: &Datamodel, introspection_result: &mut Introspecti
     {
         for model in &new_data_model.models {
             for field in &model.fields {
-                if let FieldType::Relation(info) = &field.field_type {
+                if let FieldType::Relation(new_info) = &field.field_type {
                     if let Some(old_model) = old_data_model.find_model(&model.name) {
                         for old_field in &old_model.fields {
                             if let FieldType::Relation(old_info) = &old_field.field_type {
-                                if old_info == info {
-                                    changed_relation_field_names.push((
-                                        ModelAndField {
-                                            model: model.name.clone(),
-                                            field: field.name.clone(),
-                                        },
-                                        old_field.name.clone(),
-                                    ));
+                                let other_old_field = old_data_model.find_relation_field_for_info(&old_info);
+                                if let FieldType::Relation(other_old_info) = &other_old_field.field_type {
+                                    let other_new_field = new_data_model.find_relation_field_for_info(&new_info);
+
+                                    if let FieldType::Relation(other_new_info) = &other_new_field.field_type {
+                                        if old_info == new_info && other_old_info == other_new_info {
+                                            changed_relation_field_names.push((
+                                                ModelAndField {
+                                                    model: model.name.clone(),
+                                                    field: field.name.clone(),
+                                                },
+                                                old_field.name.clone(),
+                                            ));
```

Comment:
This statement nests up to 13 levels deep (52 leading spaces) with a triple-nested loop -- I don't know much about this part of the codebase, but this seems really rough to follow. Adding a `fn new` constructor on `ModelAndField` could reduce some of the newly introduced nesting:

```rust
let mf = ModelAndField::new(&model, &field);
changed_relation_field_names.push((mf, old_field.name.clone()));
```

Reply:
I started getting rid of some of the nesting, but now also incorporated your suggestion on top